include/lsst/meas/astrom/PolynomialTransform.h

```diff
@@ -0,0 +1,253 @@
+
+    /**
+     *  Construct a new transform from existing coefficient arrays.
+     *
+     *  For both input arguments, the array element at [p, q] corresponds
+     *  to the polynomial term x^p y^q.
+     *
+     *  Both arrays are expected be square and triangular; if N is the
+     *  order of the transform, both arrays should be (N+1)x(N+1), and
+     *  elements with p + q > N should be zero.
+     */
+    PolynomialTransform(
+        ndarray::Array<double const,2,2> const & xCoeffs,
+        ndarray::Array<double const,2,2> const & yCoeffs
+    );
+
+    /**
+     *  Copy constructor.
+     *
+     *  Coefficient arrays are deep-copied.
+     */
+    PolynomialTransform(PolynomialTransform const & other);
+
+    /**
+     *  Move constructor.
+     *
+     *  Coefficient arrays are moved.
+     */
+    PolynomialTransform(PolynomialTransform && other);
+
+    /**
+     *  Copy assignment.
+     *
+     *  Coefficient arrays are deep-copied.
+     */
+    PolynomialTransform & operator=(PolynomialTransform const & other);
+
+    /**
+     *  Move constructor.
+     *
+     *  Coefficient arrays are moved.
+     */
+    PolynomialTransform & operator=(PolynomialTransform && other);
+
+    /// Lightweight swap.
+    void swap(PolynomialTransform & other);
+
+    /// Return the order of the polynomials.
+    int getOrder() const { return _xCoeffs.rows() - 1; }
+
+    /**
+     * 2-D polynomial coefficients that compute the output x coordinate.
+     *
+     * Indexing the result by [p][q] gives the coefficient of
+     * @f$x_{\mathrm{in}}^p\,y_{\mathrm{in}}^q@f$.
+     */
+    ndarray::Array<double const,2,2> getXCoeffs() const { return _xCoeffs.shallow(); }
+
+    /**
+     * 2-D polynomial coefficients that compute the output x coordinate.
+     *
+     * Indexing the result by [p][q] gives the coefficient of
+     * @f$x_{\mathrm{in}}^p\,y_{\mathrm{in}}^q@f$.
+     */
+    ndarray::Array<double const,2,2> getYCoeffs() const { return _yCoeffs.shallow(); }
+
+    /**
+     * Return an approximate affine transform at the given point.
+     */
+    afw::geom::AffineTransform linearize(afw::geom::Point2D const & in) const;
+
+    /**
+     * Apply the transform to a point.
+     */
+    afw::geom::Point2D operator()(afw::geom::Point2D const & in) const;
+
+private:
+
+    PolynomialTransform(int order);
+
+    friend PolynomialTransform compose(afw::geom::AffineTransform const & t1, PolynomialTransform const & t2);
+    friend PolynomialTransform compose(PolynomialTransform const & t1, afw::geom::AffineTransform const & t2);
+    friend class ScaledPolynomialTransformFitter;
+    friend class SipForwardTransform;
+    friend class SipReverseTransform;
+    friend class ScaledPolynomialTransform;
+
+    ndarray::EigenView<double,2,2> _xCoeffs;
+    ndarray::EigenView<double,2,2> _yCoeffs;
+    mutable Eigen::VectorXd _u; // workspace for operator() and linearize
+    mutable Eigen::VectorXd _v;
```

Comment:
Hmm.,  I think I'll just document that instance of this class should be confined to a single thread, as I think that's probably how it ought to be used.  Ultimately I expect it to be a moot point when we switch to using AST for these classes.

However, that's a reminder that the `BinomialMatrix` class I also added on this ticket is unsafe in a somewhat more worrying way, since it uses a singleton.  I'll fix that.